### PR TITLE
x-pack/filebeat/input/entityanalytics/provider/okta: allow fine-grain control of API requests

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

This adds support for specifying which of users/devices to collect from the Okta API endpoints in order to reduce network costs for users who do not need a full set of entities.

The current change does not change the behaviour of device collection of registered owners and registered users; when the "devices" dataset is selected there user entities will still be collected as they are considered here as an attribute of the device, rather than a component of the users dataset.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- For #36440

### CODE REVIEW

**Review by bhapas — COMMENTED** [review]
LGTM.

Possible to add some unit tests?

File: x-pack/filebeat/input/entityanalytics/provider/okta/okta_test.go [hunk 1]
```diff
@@ -22,156 +22,186 @@ import (
...
-		"devices": devices,
+	tests := []struct {
+		dataset     string
+		wantUsers   bool
+		wantDevices bool
+	}{
+		{dataset: "", wantUsers: true, wantDevices: true},
+		{dataset: "all", wantUsers: true, wantDevices: true},
+		{dataset: "users", wantUsers: true, wantDevices: false},
+		{dataset: "devices", wantUsers: false, wantDevices: true},
```
Comment (bhapas) [inline x-pack/filebeat/input/entityanalytics/provider/okta/okta_test.go:33]:
```suggestion
		{dataset: "USERS", wantUsers: true, wantDevices: false},
		{dataset: "DeViCes", wantUsers: false, wantDevices: true},
```

Just covering strange config to convert to lowercase

**Review by bhapas — APPROVED** [review]
LGTM

File: x-pack/filebeat/input/entityanalytics/provider/okta/okta_test.go (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/okta/okta_test.go:33]: I'm ok with not testing that. We don't specify that we are case insensitive.

**Pull request merged**